Sources/GroupByTracker/GbTracker.swift

```diff
@@ -51,11 +53,33 @@ public class GbTracker {
         self.shopperTracking = ShopperTracking(login: login, visitorID: uuid)
     }
     
+    private func setSiteFilterMetadata(metadata: [Metadata]?) -> [Metadata]? {
+        var result = metadata
+        if (result == nil) {
+            result = []
+        }
+        
+        var valueSet = false;
+        for siteFilterMetadata in result! {
+            if (siteFilterMetadata.key == "siteFilter") {
+                valueSet = true
+            }
+        }
+        
+        if (!valueSet) {
```

Comment:
This should check if !valueSet && self.siteFilterMetadataValue != nil before it adds the value.